File: Logentries/src/main/java/me/giorgirokhadze/parser/twino/NestedObjectParser.java
package me.giorgirokhadze.parser.twino;

import me.giorgirokhadze.parser.Parser;
import org.json.simple.JSONObject;

class NestedObjectParser implements Parser {

	@SuppressWarnings("unchecked")
	public JSONObject parse(String log) {
		JSONObject res = new JSONObject();
		log = log.substring(1, log.length() - 1);
		if (log.length() > 0) {
			String[] split = log.split(",");
			for (String entry : split) {
				String[] temp = entry.split("=");
				String val = null;
				if (!temp[1].equals("<null>"))
					val = temp[1];
				res.put(temp[0], val);
			}
		}
		return res;
	}

}

File: Logentries/src/main/java/me/giorgirokhadze/parser/twino/NestedObjectParser.kt
package me.giorgirokhadze.parser.twino

import me.giorgirokhadze.parser.Parser
import org.json.simple.JSONObject

internal class NestedObjectParser : Parser {

    override fun parse(line: String): JSONObject {
        @Suppress("NAME_SHADOWING")
        var line = line
        val result = JSONObject()
        if (line.length > 0) {
            line = line.substring(1, line.length - 1)
            while (line.length > 0) {
                val key = line.substring(0, line.indexOf('='))
                val value: String
                if (line[line.indexOf('=') + 1] == '[') {
                    value = line.substring(line.indexOf('=') + 1, line.indexOf(']') + 1)
                    line = line.substring(line.indexOf(']') + 1)
                } else {
                    value = if (line.indexOf(',') < 0) line.substring(line.indexOf('=') + 1, line.length) else line.substring(line.indexOf('=') + 1, line.indexOf(','))
                    line = if (line.indexOf(',') == -1) "" else line.substring(line.indexOf(',') + 1)
                }
                //noinspection unchecked
                result.put(key, if (value == "<null>") null else value)

            }
        }
        return result
    }

}
